addons/im_livechat/controllers/cors/attachment.py

```diff
@@ -0,0 +1,22 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.http import route
+from odoo.addons.mail.controllers.attachment import AttachmentController
+from odoo.addons.im_livechat.tools.misc import force_guest_env
+
+
+class LivechatAttachmentController(AttachmentController):
+    @route("/im_livechat/cors/attachment/upload", auth="public", cors="*", csrf=False)
+    def im_livechat_attachment_upload(self, guest_token, ufile, thread_id, thread_model, is_pending=False, **kwargs):
+        force_guest_env(guest_token)
+        return self.mail_attachment_upload(ufile, thread_id, thread_model, is_pending, **kwargs)
+
+    @route("/im_livechat/cors/attachment/delete", methods=["POST"], type="json", auth="public", cors="*")
+    def im_livechat_attachment_delete(self, guest_token, attachment_id, access_token=None):
+        force_guest_env(guest_token)
+        return self.mail_attachment_delete(attachment_id, access_token)
+
+    def _should_generate_attachment_access_token(self, thread):
+        return super()._should_generate_attachment_access_token(thread) or (
+            thread._name == "discuss.channel" and thread.channel_type == "livechat"
+        )
```

Comment:
I'm still not sure why we would need that. Shouldn't it go through the `discuss/channel` attachment route with the cors+guest token?

Reply:
There is no way to attach the guest token since the route is triggered from image `src`.

Reply:
Isn't it possible to make a livechat specific version of the route supporting the param? As it's a get route it can just be added inside the url as `&token=`

Reply:
I don't like the idea of sending the guest token as a query parameter since it will end up in several places (logs, system information...). What do you think?